files/preStudies/backends.tex

```diff
@@ -0,0 +1,58 @@
+\section{Backends}
+
+This section will evaluate different options to support the project with data storage and possibly for running heavy jobs that you do not what the clients to run. Some \textit{must have} and \textit{nice to have} capabilities will be listed below.
+	
+\textbf{Must have:}
+	\begin{itemize}
+	\item Be able to store data
+	\item Support authentication.
+	\end{itemize}
+	
+\textbf{Nice to have:}
+	
+	\begin{itemize}
+	\item Generic API
+	\item Reactive data. Updated data in the database should reflect on the client.
+	\end{itemize}
+	
+
+\subsubsection{Firebase}
+
+\textbf{Function:} Data storage
+\\
+\textbf{Database type:} NoSQL (JSON)
+\\
+\textbf{Description:}
+\\
+Firebase is a database platform from \textit{Google}. It has a lot of features you do not normally see on a database. One of the important things is authorization. Usually stuff like authorization goes through some sort of Web API before access is allowed to the database. Firebase can handle this aspect without a Web API to parent it. Firebase is a real-time database, which means that the data is reactive. So whenever there is a change in the database, that change is pushed to the clients to keep them up to date at all times. Pushing data into the database is done with JSON. When using Firebase you can not run functions or jobs like in other systems. But a company named \textit{Zapier} have made this available through a product of theirs. This is an extra cost and might not be needed, But it is nice to know that there is an option for this.
+
+
+\subsubsection{Apollo}
+
+\textbf{Function:} Reactive Data storage
+\\
+\textbf{Database type:} None specific, free to chooose. Client application interface is GraphQL.
+\\
+\textbf{Description:}
+\\
+Apollo or the Apollostack is a modern data stack that offers reactive data. This means that your data will always be up-to-date, if you want it to be that is. You can choose your poise. If you do not need reactive data you can choose not to. Apollo is a data stack provided by the \textit{Meteor} team. As opposed to \textit{Meteor}, Apollo is ment to be useable in any application. There are integration guides for Angular 2, Redux, Meteor, React and React native. But you should be able to integrate it into any javascript frontend. Apollo uses GraphQL, which consists of 2 parts. A Schema part that tells what type of data is available on the server and a query language that enabled the client to describe the data it needs. This means that the client have no idea what type of database is behind GraphQL. It is comparable to REST, but where you very clearly define what you need instead of getting an object like you usually would when using REST.
```

Comment:
Apollo in italics all over.
it to be that is -> it to be, that is.
Angular2, Redux, Metoer, React, React Native in italics.
No dot before the "but"s.
javascript -> JavaScript
GraphQL in italics.
enabed -> enables
The last phrase is a bit pleonastic. Maybe just mention that you define what properties you want as compared to a typical REST.